api/src/main/java/org/openmrs/api/db/hibernate/PersonLuceneQuery.java

```diff
@@ -112,6 +112,28 @@ public PersonLuceneQuery(SessionFactory sessionFactory) {
 	 * @param gender the gender of the person to search  
 	 * @return the LuceneQuery that returns Persons with a soundex representation of the firstName and other defined search criteria
 	 */
+	public LuceneQuery<PersonName> getSoundexPersonNameSearchOnThreeNames(String n1, String n2, String n3,  Integer birthyear, boolean includeVoided, String gender) {
+		String threeNameQuery = "";
+		String givenNameMatchCondition = "(givenNameSoundex:n1^6 OR givenNameSoundex:n2^2 OR givenNameSoundex:n3)";
+		String middleNameMatchCondition = "(middleNameSoundex:n1^2 OR middleNameSoundex:n2^6 OR middleNameSoundex:n3^1)";
+		String familyNameMatchCondition = "(familyNameSoundex:n1^1 OR familyNameSoundex:n2^2 OR familyNameSoundex:n3^6)";
+		String familyName2MatchCondition = "(familyName2Soundex:n1^1 OR familyName2Soundex:n2^2 OR familyName2Soundex:n3^6)";
+		
+		threeNameQuery = givenNameMatchCondition+ " OR " + middleNameMatchCondition + " OR " + familyNameMatchCondition + " OR " + familyName2MatchCondition;
+ 		threeNameQuery = "(" +  threeNameQuery.replace("n1", n1).replace("n2", n2).replace("n3", n3) + ")";
```

Comment:
It seems to me we should at least run `LuceneQuery.escapeQuery()` on these parameters. But is there a way we could do this more cleanly, i.e., by passing these as paramters instead of a string replace operation?

If not, we should at least have this query stored as a constant string since we really just need to do some replacements right before it's executed.

Reply:
What I could do is to directly replace in the query definition the n1, n2, n3 parameter with the correct value but this would made the code less readable I guess.

I think I would prefer to make it a constant and then replacing the values. 

Can you elaborate a bit more on what your thoughts are on "passing these as parameters"? 

Anyways, the escapeQuery can be added easily. I will probably do that till next Sunday!
Thanks!